packages/navitem/src/react/horz.tsx

```diff
@@ -0,0 +1,74 @@
+import { CaretDownIcon } from '@pluralsight/ps-design-system-icon'
+import { PropsOf } from '@pluralsight/ps-design-system-util'
+import { compose, css } from 'glamor'
+import React, { useContext } from 'react'
+
+import stylesheet from '../css'
+
+import Context from './context'
+import { Bar, Button } from './common'
+
+const styles = {
+  container: () => css(stylesheet['.psds-navitem__horz-container']),
+
+  caret: () => css(stylesheet['.psds-navitem__horz-caret']),
+
+  icon: (props: { hasLabel: boolean }) =>
+    compose(
+      css(stylesheet['.psds-navitem__horz-icon']),
+      !props.hasLabel && css(stylesheet['.psds-navitem__horz-icon--icon-only'])
+    ),
+
+  label: () => css(stylesheet['.psds-navitem__horz-label']),
+
+  layout: (props: { menu?: unknown }) =>
+    compose(
+      css(stylesheet['.psds-navitem__horz-layout']),
+      props.menu && css(stylesheet['.psds-navitem__horz-layout--menu'])
+    )
+}
+
+export const HorzLayout: React.FC = props => {
+  const { icon, menu } = useContext(Context)
+  const hasLabel = !!props.children
+
+  return (
+    <HorzContainer>
+      <Button>
+        <span {...styles.layout({ menu })}>
+          {icon && <HorzIcon hasLabel={hasLabel}>{icon}</HorzIcon>}
+          {hasLabel && <HorzLabel>{props.children}</HorzLabel>}
+          {menu && <HorzCaret />}
+        </span>
+      </Button>
+
+      <Bar />
+    </HorzContainer>
+  )
+}
+HorzLayout.displayName = 'NavItem.HorzLayout'
+
+const HorzContainer: React.FC<PropsOf<'span'>> = props => {
```

Comment:
Maybe I'm scared of a boogie man who doesn't exist, but I really don't want to use PropsOf except where it's deemed necessary.  

At first glance, PropsOf usage here seems much wider than necessary, for a couple unrelated reasons:

1. React.HTMLAttributes has gotten us a long way -- almost through the TS conversion of the entire DS codebase.  It feels uneven to change the strategy on these last couple packages.

2. `HorzContainer` simply doesn't need this.  It doesn't take props, so it doesn't need to be typed to anticipate HTML attributes.

Props are able to be sent:

```
51:   const HorzContainer: React.FC<PropsOf<'span'>> = props => {
```

, but they are never sent:

```
36:   <HorzContainer>
```

I don't think we need this prop spreading or even the props param.

Reply:
1: i've addressed above around PropsOf/JSX.IntrinsicElements providing a more complete set of typings and more accurate introspection. at this point i'm reasonably sure(75% 😉 ) that JSX.IntrinsicElements doesn't provide any complier issues when used to type a single element or a small set of elements for polymorphism. It is also used in the react types everything we reach for `ComponentProps` or `ElementType` which we have used with not apparent downsides. I agree that it's a pivot from our previous strategy of using `HTMLAttributes` but at this point i think it's the write pivot to make - especially for the public apis of the components

2: i see where you're coming from since the props aren't used and the components are internal. I personally like spreading props for all valid attributes - and with typescript we benefit from the type checking to ensure the props we pass are valid. but it's a personal preference and not something i'm all that attached to. i'd be happy to add typings as we need them to internal components

as always i think the discussion is important and happy to keep churning on it

Reply:
1: Thanks for your reasoning around using JSX.IntrinisicElements. I agree with this being a pivot.  Let me play with it a little today to further convince myself (hopefully approach a happy 75% with you) that this is a good thing.

2: Not a huge deal.  The most basic thing I'm saying is that if the props aren't used, don't type them and don't spread them.  This preference is in favor of simplicity and noise reduction.